components/server/src/memory.rs

```diff
@@ -13,15 +13,32 @@ pub struct MemoryTraceManager {
 
 impl MemoryTraceManager {
     pub fn flush(&mut self, _now: Instant) {
+        let raft_router = "raft_router".to_string();
+        let apply_router = "apply_router".to_string();
         for provider in &self.providers {
             let provider_name = provider.name();
             let ids = provider.get_children_ids();
             for id in ids {
                 let sub_trace = provider.sub_trace(id);
                 let sub_trace_name = sub_trace.name();
-                MEM_TRACE_SUM_GAUGE
-                    .with_label_values(&[&format!("{}-{}", provider_name, sub_trace_name)])
-                    .set(sub_trace.sum() as i64)
+                if sub_trace_name == raft_router || sub_trace_name == apply_router {
+                    let alive = sub_trace.sub_trace(Id::Name("alive"));
```

Comment:
Can we make it more generic, like using `get_children_ids` to record sub-traces recursively?

Reply:
Good point.